hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMAssignUserToTenantRequest.java

```diff
@@ -325,21 +324,10 @@ public OMClientResponse validateAndUpdateCache(
           new CacheKey<>(accessId),
           new CacheValue<>(Optional.of(roleName), transactionLogIndex));
 
-      // Add to S3SecretTable.
-      // Note: S3SecretTable will be deprecated in the future.
+      // Add S3SecretTable cache entry
       acquiredS3SecretLock = omMetadataManager.getLock()
           .acquireWriteLock(S3_SECRET_LOCK, accessId);
 
-      // Expect accessId absence from S3SecretTable
-      // TODO: This table might be merged with tenantAccessIdTable later.
-      if (omMetadataManager.getS3SecretTable().isExist(accessId)) {
-        LOG.error("accessId '{}' already exists in S3SecretTable", accessId);
-        throw new OMException("accessId '" + accessId +
-            "' already exists in S3SecretTable",
-            OMException.ResultCodes.INVALID_REQUEST);
-      }
```

Comment:
Keep this but instead of throwing, print a warning (to the OM log) if the accessId already exists in the `getS3SecretTable`?

Reply:
Oh yeah let me put this back in. I think I mistakenly thought this block was operating on the old multi-tenant s3 secret. I think we do want the exception if the access ID exists already though so we don't silently overwrite someone else's access.